saged/utils.py

```diff
@@ -54,6 +55,93 @@
               ]
 
 
+def remove_study_samples(dataset: "datasets.ExpressionDataset",
+                         studies_to_remove: Set[str]) -> "datasets.ExpressionDataset":
+    """
+    Remove the samples corresponding to the given studies from a dataset
+
+    Arguments
+    ---------
+    dataset - The dataset samples will be removed from
+    studies_to_remove - The studies whose samples should be removed from the dataset
+
+    Returns
+    -------
+    dataset - The dataset after samples have been removed
+    """
+    samples = dataset.get_samples()
+    sample_to_study = dataset.get_samples_to_studies()
+    samples_to_remove = get_samples_in_studies(samples,
+                                               studies_to_remove,
+                                               sample_to_study)
+    dataset.remove_samples(samples_to_remove)
+
+    return dataset
+
+
+def split_by_tissue(data: "datasets.ExpressionDataset",
+                    tissues: List[str],
+                    num_splits: int = 5) -> List["datasets.ExpressionDataset"]:
+    """
+    Split a dataset into sections, with each split containing distinct tissues
+
+    Arguments
+    ---------
+    data: The dataset to split
+    tissues: The list of tissues to keep
+    num_splits: The number of partitions to divide the dataset into
+
+    Returns
+    -------
+    splits: A list of datasets split by tissue
+    """
+    if num_splits <= 1:
+        return data
+
+    splits = []
+
+    # Split labels into lists
+    all_subsets = []
+    for i in range(num_splits):
+        subset_tissues = [t for t_index, t in enumerate(tissues) if t_index % num_splits == i]
+        all_subsets.append(subset_tissues)
+
+    # Use subset_to_label to create five datasets
+    for subset in all_subsets:
+        data = data.subset_samples_to_labels(subset)
+        # This is very memory intensive since the full dataset gets stored behind the scenes after
+        # subsetting. It could be made more efficient with a special subset function that
+        # removes the old data, but that may make the API too messy
+        data_copy = copy.deepcopy(data)
+        data.reset_filters()
+
+        splits.append(data_copy)
+
+    # Ensure no studies are shared between the datasets
```

Comment:
Are there a lot of studies that contain samples from multiple tissues?

Reply:
There are a decent amount, iirc I lose ~10 percent of the data by making sure that the studies are distinct across splits